scripts/analysis.py

```diff
@@ -250,229 +116,198 @@ def get_sim_time_duration(cursor):
     timestep: list
         timeseries up to duration
     """
-    cur = cursor
-
-    info = cur.execute('SELECT initialyear, initialmonth,'
-                       + ' duration FROM info').fetchone()
+    info = cursor.execute('SELECT initialyear, initialmonth, '
+                          'duration FROM info').fetchone()
     init_year = info[0]
     init_month = info[1]
     duration = info[2]
-    timestep = np.linspace(0, info[2]-1, num=info[2])
+    timestep = np.linspace(0, info[2] - 1, num=info[2])
 
     return init_year, init_month, duration, timestep
 
 
-def isotope_mass_time_list(resources, compositions):
-    """Creates an list with isotope name, mass, and time
-
-    Parameters
-    ----------
-    resources: list
-        resource data from the resources table
-    compositions: list
-        composition data from the compositions table
-
-    Returns
-    -------
-    list
-        isotope name list
-    list
-        isotope mass list
-    list
-        isotope transaction time list
-
-    """
-
-    temp_isotope = []
-    temp_mass = []
-    time_list = []
-    for res in resources:
-        res_qualid = res[2]
-        indices = [x for x, y in enumerate(compositions) if y[0] == res_qualid]
-        for index in indices:
-            nucid = compositions[index][1]
-            mass_frac = compositions[index][2]
-            mass_waste = res[0]
-            res_time = res[1]
-            temp_isotope.append(nucid)
-            temp_mass.append(mass_frac*mass_waste)
-            time_list.append(res_time)
-
-    return temp_isotope, temp_mass, time_list
-
-
-def plot_in_out_flux(cursor, facility, influx_bool, title, outputname):
-    """plots timeseries outflux from facility name in kg.
+def get_timeseries(in_list, duration, multiplyby, cumulative):
+    """ returns a timeseries list of a given data
 
     Parameters
     ----------
-    cursor: sqlite cursor
-        sqlite cursor
-    facility: str
-        facility name
-    influx_bool: bool
-        if true, calculates influx,
-        if false, calculates outflux
-    title: str
-        title of the multi line plot
-    outputname: str
-        filename of the multi line plot file
+    in_list: list
+        list of data to be created into timeseries
+        list[0] = time
+        list[1] = value, quantity
+    duration: int
+        duration of the simulation
+    multiplyby: int
+        integer to multiply the value in the list by for
+        unit conversion from kilograms
```

Comment:
from kilograms to what? If tons, why not call this variable kg_to_tons ? If you want it to remain generic, then unit convert in a different function. Don't do 10 things at once. 
Also, please note the existence of a unit parsing library in python that may be of interest : https://pint.readthedocs.io/en/0.7.2/